pytest-embedded-wokwi/pytest_embedded_wokwi/wokwi_cli.py

```diff
@@ -0,0 +1,85 @@
+import json
+import os
+import typing as t
+from pathlib import Path
+
+from pytest_embedded import __version__
+from pytest_embedded.log import DuplicateStdoutPopen
+
+if t.TYPE_CHECKING:
+    from pytest_embedded_idf.app import IdfApp
+
+
+target_to_board = {
+    'esp32': 'board-esp32-devkit-c-v4',
+    'esp32c3': 'board-esp32-c3-devkitm-1',
+    'esp32c6': 'board-esp32-c6-devkitc-1',
+    'esp32h2': 'board-esp32-h2-devkitm-1',
+    'esp32s2': 'board-esp32-s2-devkitm-1',
+    'esp32s3': 'board-esp32-s3-devkitc-1',
+}
+
+
+class WokwiCLI(DuplicateStdoutPopen):
+    """
+    WokwiCLI class
+    """
+
+    SOURCE = 'Wokwi'
+
+    WOKWI_CLI_PATH = 'wokwi-cli'
+
+    def __init__(
+        self,
+        wokwi_cli_path: t.Optional[str] = None,
+        app: t.Optional['IdfApp'] = None,
+        **kwargs,
+    ):
+        """
+        Args:
+            wokwi_cli_path: Wokwi CLI arguments
+        """
+        self.app = app
+        flasher_args = Path(app.binary_path, 'flasher_args.json')
+
+        with open(os.path.join(app.app_path, 'wokwi.toml'), 'wt') as f:
```

Comment:
Is it possible to keep the auto-generated files in build directory, instead? (Or at least to check that the file exists and not overwrite it if it does.)

Another question, can we add an option to specify the TOML file? I imagine that for a project supporting multiple boards (e.g. ESP-BSP) we would want to use `@pytest.mark.parametrize` to specify different TOML files depending on the board.

Reply:
I'm trying to figure out what's the best way to handle this. The tradeoff is between ease of getting started (minimal configuration), and compatibility with the VS Code plugin. 

The VS Code plugin expects both wokwi.toml and diagram.json to be in the same directory (you could have multiple directories with multiple configurations). 

For basic CLI usage, you could probably get away with just specifying the type of the target type and have the wokwi.toml / diagram.json generated as we do above, without committing it to the repo. This makes switching to a different target super easy, as pytest already knows about the target and can generate diagram.json as we're doing now. @SergioGasquez also mentioned to me he'd love to see something similar in the CLI (a flag that generates diagram.json given an MCU or so).

But once you want to get more out of the simulator, you'll probably want to commit diagram.json and possibly also wokwi.toml (e.g. if you use some [custom chips](https://docs.wokwi.com/vscode/project-config#custom-chips). In more complex use cases, switching to a different MCU probably means you need to adjust the connections anyway, so you'd want to have multiple diagram configurations, as you suggested.

So I'm really not sure how to go about this.

Reply:
What about supporting extra arguments in wokwi CLI:

* `--config=path` — specifies the config file, if not specified then wokwi.toml in the current working directory is used
* `--diagram=path` — specifies the diagram file, if not specified then diagram.json in the current working directory is used

pytest-embedded-qemu can then specify have config options to specify these files (including pytest.mark.parametrize), and this will still work in VS Code projects.

Reply:
As a first step, we no longer override wokwi.toml or diagram.json if they already exist. 

If wokwi.toml already exist, we check if the firmware / elf paths are set correctly, and if they are not, we update them while keeping the rest of the config intact.

For diagram.json, if it already exists, we read it and check if the selected board matches the current target, and if it does not, we print a warning.